### PR TITLE
[DebugInfo] Corrected debuginfo for modules

### PR DESCRIPTION
**Pull request by bhuvanendrakumarn**
This PR is same as https://github.com/flang-compiler/flang/pull/898/ to support modules debug info. i recreated this fresh PR for the same fix. please note 898 has the approval of @bryanpkc already.

### CODE REVIEW

**Comment by bryanpkc** [comment]
Please squash the commits and add "[DebugInfo]" prefix to the commit title. It also wouldn't hurt to add more explanation in the commit message.

**Comment by bhuvanendrakumarn** [comment]
> Please squash the commits and add "[DebugInfo]" prefix to the commit title. It also wouldn't hurt to add more explanation in the commit message.

thanks, its done now

**Comment by kiranchandramohan** [comment]
@bhuvanendrakumarn Can you check the CI failure?

**Comment by bhuvanendrakumarn** [comment]
> @bhuvanendrakumarn Can you check the CI failure?

CI checks are passing now

File: tools/flang2/flang2exe/ll_write.cpp [hunk 1]
```diff
@@ -1027,6 +1027,18 @@ static const MDTemplate Tmpl_DIModule[] = {
...
+static const MDTemplate Tmpl_DIModule_11[] = {
+  { "DIModule", TF, 5 },
+  { "tag",                      DWTagField,    FlgHidden },
+  { "scope",                    NodeField,     0 },
+  { "name",                     StringField,   0 },
+  { "file",                     NodeField,     0 },
+  { "line",                     UnsignedField, 0 },
+  //{ "configMacros",           StringField, FlgOptional },
+  //{ "includePath",            StringField, FlgOptional },
+  //{ "isysroot",               StringField, FlgOptional },
```
Comment (kiranchandramohan) [inline tools/flang2/flang2exe/ll_write.cpp:1039]: Nit:Can you remove the commented code?

Comment (bhuvanendrakumarn) [reply tools/flang2/flang2exe/ll_write.cpp]: thanks, its addressed

**Review by kiranchandramohan — APPROVED** [review]
LGTM

**Comment by bhuvanendrakumarn** [comment]
@bryanpkc, could you please approve this PR?

**Review by bryanpkc — APPROVED** [review]

**Comment by bhuvanendrakumarn** [comment]
Thanks for the review and approvals @bryanpkc @kiranchandramohan

**Pull request merged**